build_tools/azure/install.sh

```diff
@@ -120,6 +130,20 @@ python_environment_install() {
         pip install https://github.com/joblib/joblib/archive/master.zip
         echo "Installing pillow master"
         pip install https://github.com/python-pillow/Pillow/archive/main.zip
+    elif [[ "$DISTRIB" == "pip-nogil" ]]; then
+        setup_ccache  # speed-up the build of CPython it-self
+        ORIGINAL_FOLDER=`pwd`
+        cd ..
+        git clone --depth 1 https://github.com/colesbury/nogil
+        cd nogil
+        ./configure && make -j 2
+        ./python -m venv $ORIGINAL_FOLDER/$VIRTUALENV
+        cd $ORIGINAL_FOLDER
+        source $VIRTUALENV/bin/activate
+
+        python -m pip install -U pip
+        echo "Installing build dependepencies with pip from the nogil repository"
+        pip install numpy scipy cython joblib threadpoolctl
```

Comment:
I think we need to set the index to https://d1yxz45j0ypngg.cloudfront.net/ ?

Reply:
It's done automatically on my laptop.

Reply:
I confirm this happens automatically on the CI as well:

https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=41178&view=logs&j=67fbb25f-e417-50be-be55-3b1e9637fce5&t=d636361e-5b30-5c70-56e1-c2011256a577&l=1625

The pip provided with nogil CPython is patched to use this index by default.